src/MarkdownWriter/CommandHelpMarkdownWriter.cs

```diff
@@ -0,0 +1,235 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using Microsoft.PowerShell.PlatyPS.Model;
+
+[assembly: System.Runtime.CompilerServices.InternalsVisibleTo("Microsoft.PowerShell.PlatyPS.Tests")]
+namespace Microsoft.PowerShell.PlatyPS.MarkdownWriter
+{
+    /// <summary>
+    /// Write the CommandHelp object to a file in markdown format.
+    /// </summary>
+    internal class CommandHelpMarkdownWriter
+    {
+        private readonly string _filePath;
+        private StringBuilder? sb = null;
+        private readonly Encoding _encoding;
+
+        /// <summary>
+        /// Initialize the writer with settings.
+        /// </summary>
+        /// <param name="settings">Settings needs for the markdown writer.</param>
+        public CommandHelpMarkdownWriter(MarkdownWriterSettings settings)
+        {
+            string path = settings.DestinationPath;
+
+            if (string.IsNullOrEmpty(path))
+            {
+                throw new ArgumentNullException(nameof(path));
+            }
+            else
+            {
+                _filePath = path;
+                _encoding = settings.Encoding;
+            }
+        }
+
+        /// <summary>
+        /// Write the CommandHelp object to the specified path in the the settings during initialization.
+        /// The file is overwritten and not appended as we do not expect more than one CommandHelp per file.
+        /// </summary>
+        /// <param name="help">CommandHelp object to write.</param>
+        /// <param name="noMetadata">Specify if the metadata header should be written.</param>
+        /// <param name="metadata">Additional metadata that is appended to the standard metadata header. If nometata is true, additional metadata is not written.</param>
+        /// <returns>FileInfo object of the created file</returns>
+        internal FileInfo Write(CommandHelp help, bool noMetadata, Hashtable? metadata = null)
+        {
+            sb ??= new StringBuilder();
```

Comment:
If we move this in ctor and make `sb` non-nullable we have no need to check for null in every line like `sb?.AppendLine(Constants.YmlHeader);`.

Reply:
Fixed by using a stringbuilder pool and initializing it at declaration